restorer.go

```diff
@@ -62,10 +75,12 @@ func (res *Restorer) restoreTo(dst string, dir string, treeID backend.ID) error
 				}
 			}
 
-			// Restore directory timestamp at the end. If we would do it earlier, restoring files within
-			// the directory would overwrite the timestamp of the directory they are in.
-			if err := node.RestoreTimestamps(filepath.Join(dst, dir, node.Name)); err != nil {
-				return err
+			if selectedForRestore {
```

Comment:
That's an interesting corner case, we can leave it for now (just don't restore metadata for dirs that aren't selected for restore), but I think we can add an option for it later, so that the user can decide to restore the metadata of directories restored implicitely because they contain restored files.